generator/document.go

```diff
@@ -168,46 +166,71 @@ func RandomString(n int) string {
 	return string(s)
 }
 
-func GeneratePatches(num_patch int, c chan map[string]interface{}) ([]interface{}, error) {
-	ids_to_patch := genUniqueInRange(getMaxDoc(), num_patch)
+func GeneratePatches(num_patch int, destination string, c chan map[string]interface{}) ([]interface{}, error) {
 	patches := make([]interface{}, 0)
 
+	ids_to_patch := genUniqueInRange(getMaxDoc(), num_patch)
 	for _, id := range ids_to_patch {
-		patch := generatePatch(id, <-c)
-		patches = append(patches, patch)
-	}
+		if (destination == "elastic") {
+			patch := generateElasticPatch(id, <-c)
+			patches = append(patches, patch)
 
+		} else if (destination == "rockset") {
+			patch := generateRocksetPatch(id, <-c)
+			patches = append(patches, patch)
+		}
+	}
 	return patches, nil
 }
 
-func RandomFieldAdd(c chan map[string]interface{}) {
+func RandomFieldAdd(destination string, c chan map[string]interface{}) {
 	// Adding fields or array members
 	for {
-		options := []map[string]interface{}{{
-			"op":    "add",
-			"path":  "/" + faker.UUIDDigit(),
-			"value": faker.Email(),
-		},
-			{
+		if (destination == "rockset") {
+			options := []map[string]interface{}{{
+				"op":    "add",
+				"path":  "/" + faker.UUIDDigit(),
+				"value": faker.Email(),
+				},
+				{
 				"op":    "add",
 				"path":  "/Tags/-",
 				"value": faker.UUIDHyphenated(), // Append to tags array
-			},
+				},
+			}
+			shuffleAndFillChannel(options, c)
+		} else if (destination == "elastic") {
+			options := []map[string]interface{}{{
+					"doc": map[string]interface{}{
+						faker.UUIDDigit(): faker.Email(),
+						"_ts": CurrentTimeMicros(),
+					},
+				},
+				{
+					"script": map[string]interface{}{
+						"source": "ctx._source.Tags.add(params.tag)",
+						"params": map[string]interface{}{
+							"tag": faker.UUIDHyphenated(),
+							"_ts": CurrentTimeMicros(),
```

Comment:
Is this `_ts` being used anywhere?

Reply:
It was supposed to be used to get latency for updates but I see we changed the query for rockset a while back https://github.com/rockset/rockbench/commit/c1efc10366ea0029a0248f8992da83d0c6f6d461#diff-063dbbcaf9565d18aa7565646fd28529e416d522a741f5b49d508bd8d119f97c. I will keep the query for elastic the same and we can later revisit how to get latency for updates